python/whylogs/experimental/api/constraints/__init__.py

```diff
@@ -0,0 +1,69 @@
+from typing_extensions import TypedDict
+from typing import Optional, Union, List
+from whylogs.core.constraints.factories import no_missing_values
+from whylogs.core.constraints.factories import is_in_range
+from whylogs.core.constraints.factories import column_is_probably_unique
+from logging import getLogger
+from whylogs.core.constraints.metric_constraints import Constraints, MetricConstraint, DatasetConstraint
+import yaml
+
+logger = getLogger(__name__)
+
+
+constraints_mapping = {
+    "no_missing_values": no_missing_values,
+    "is_in_range": is_in_range,
+    "column_is_probably_unique": column_is_probably_unique,
+}
+
+
+def assemble_constraint(constraint_dict: dict) -> Optional[TypedDict]:
+    constraint_name = None
```

Comment:
I think we could handle prefix expression constraints with something like
```
if "expression" in constraint_dict:  # alternative to factory key
  condition = PrefixCondition(constraint_dict.get("expression"))
  ...
elif "factory" in constraint_dict:
  # as below
```

Reply:
If `expression` is present, constraint will be assembled using it, provided it has a name.